helix-term/src/ui/picker.rs

```diff
@@ -390,13 +407,15 @@ pub struct Picker<T: Item> {
     show_preview: bool,
 
     callback_fn: Box<dyn Fn(&mut Context, &T, Action)>,
+    on_close_fn: Box<dyn Fn(&mut Context)>,
 }
 
 impl<T: Item> Picker<T> {
     pub fn new(
         options: Vec<T>,
         editor_data: T::Data,
         callback_fn: impl Fn(&mut Context, &T, Action) + 'static,
+        on_close_fn: impl Fn(&mut Context) + 'static,
```

Comment:
I added this function so that we can run it when the picker is dismissed but since it is not needed in the other context, I don't know if there is a better way

Reply:
I think this mostly works fine but I think you could merge these two callbacks here and pass `None` when the picker is closed:

```rust
callback_fn: impl Fn(&mut Context, &T, Option<Action>) + 'static,
```

Reply:
So `callback_fn` should be invoked here with `None` just before `close_fn` ? 
https://github.com/helix-editor/helix/blob/b6331394a3f341ad21f8fad3e6e0b93becda9ce5/helix-term/src/ui/picker.rs#L619-L621